### PR TITLE
Test scenarios for audit converted to templates

### PR DESCRIPTION
**Pull request by marcusburghardt**
#### Description:

Where it was possible, individual test scenarios in audit rules were moved to templates.

#### Rationale:

This reduces duplicated code, the number of files and makes it simpler to be managed.

### CODE REVIEW

**Comment by jan-cerny** [comment]
:pleading_face: 
/reset

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_unsuccessful_file_modification_open/tests/audit_o_creat_rules.fail.sh [hunk 1]
```diff
@@ -1,5 +0,0 @@
-#!/bin/bash
-
-# remediation = none
-
-cp $SHARED/audit_open_o_creat.rules /etc/audit/rules.d/
```
Comment (jan-cerny) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_unsuccessful_file_modification_open/tests/audit_o_creat_rules.fail.sh:5]: Is this test replaced by anything?

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_unsuccessful_file_modification_creat/tests/correct_rules.pass.sh [hunk 2]
```diff
@@ -1,9 +0,0 @@
-#!/bin/bash
-#
-
-rm -f /etc/audit/rules.d/*
-> /etc/audit/audit.rules
-echo "-a always,exit -F arch=b32 -S creat -F exit=-EACCES -F auid>=1000 -F auid!=unset -F key=access" >> /etc/audit/rules.d/access.rules
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_unsuccessful_file_modification_creat/tests/correct_rules.pass.sh:6]: Hi Jan, the `key` filter on audit is a free field. I mean that the administrator can choose everything they want to identify the affected event in the logs without any practical effect in the rule itself. Due that, even the OVAL instruction is to ignore the values of `key`. So, we are fine here for the test scenario as far a "key" value is defined.

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_unsuccessful_file_modification_open/tests/audit_o_creat_rules.fail.sh (hunk 1 above)
Comment (marcusburghardt) [reply linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_unsuccessful_file_modification_open/tests/audit_o_creat_rules.fail.sh:5]: Yes. The templated tests cover the same intention of this file.

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_unsuccessful_file_modification_creat/tests/correct_rules.pass.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (jan-cerny) [reply linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_unsuccessful_file_modification_creat/tests/correct_rules.pass.sh:6]: Thanks for explanation!

**Review by jan-cerny — APPROVED** [review]

**Pull request merged**